### PR TITLE
GEODE-6211: Add dotnet transaction example

### PR DESCRIPTION
**Pull request by pdxcodemonkey**
Co-authored-by: Ivan Godwin <igodwin@pivotal.io>

### CODE REVIEW

File: examples/dotnet/CMakeLists.txt.in [hunk 1]
```diff
@@ -25,4 +25,5 @@ add_subdirectory(PdxAutoSerializer)
 add_subdirectory(PdxSerializableCs)
 add_subdirectory(PutGetRemove)
 add_subdirectory(RemoteQueryCs)
+add_subdirectory(TransactionCs)
```
Comment (pivotal-jbarrett) [inline examples/dotnet/CMakeLists.txt.in]: What does `Cs` mean?

Comment (igodwin) [reply examples/dotnet/CMakeLists.txt.in]: @pivotal-jbarrett , it's C#. Not sure how it started.

Comment (pdxcodemonkey) [reply examples/dotnet/CMakeLists.txt.in]: Turns out, you can’t have two projects in the same VS solution whose names only differ by case.  We (I) adopted the convention so we could still build all examples in the same solution.

Comment (pdxcodemonkey) [reply examples/dotnet/CMakeLists.txt.in]: The configure and install targets use paths based on the cmake variable ADD_EXAMPLE_NAME, so the names have to match for now.  We could extricate the two things from each other, but that's IMO work beyond the scope of this PR.

File: examples/dotnet/TransactionCs/startserver.ps1 [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    else
+    {
+        $GFSH_PATH = "$env:GEODE_HOME\bin\gfsh.bat"
+    }
+}
+
+if ($GFSH_PATH -ne "")
+{
+   Invoke-Expression "$GFSH_PATH -e 'start locator --name=locator --dir=$PSScriptRoot\locator' -e 'start server --name=server --dir=$PSScriptRoot\server' -e 'create region --name=exampleRegion --type=PARTITION'"
+}
```
Comment (pivotal-jbarrett) [inline examples/dotnet/TransactionCs/startserver.ps1]: Missing terminating new line.

File: examples/dotnet/TransactionCs/stopserver.ps1 [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    else
+    {
+        $GFSH_PATH = "$env:GEODE_HOME\bin\gfsh.bat"
+    }
+}
+
+if ($GFSH_PATH -ne "")
+{
+   Invoke-Expression "$GFSH_PATH -e 'connect' -e 'destroy region --name=exampleRegion' -e 'stop server --name=server' -e 'stop locator --name=locator'"
+}
```
Comment (pivotal-jbarrett) [inline examples/dotnet/TransactionCs/stopserver.ps1]: Missing terminating new line.

**Pull request merged**